pkg/template/diff_printer.go

```diff
@@ -0,0 +1,180 @@
+package template
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+)
+
+type DiffPrinter struct {
+	w      io.Writer
+	writer *tablewriter.Table
+
+	useColor bool
+	title    string
+
+	appendCalls int
+	headerLen   int
+}
+
+func NewDiffPrinter(w io.Writer, hasColor, hasBorder bool) *DiffPrinter {
+	wr := tablewriter.NewWriter(w)
+	wr.SetBorder(hasBorder)
+	wr.SetRowLine(hasBorder)
+
+	return &DiffPrinter{
+		w:        w,
+		writer:   wr,
+		useColor: hasColor,
+	}
+}
+
+func (d *DiffPrinter) Render() {
+	if d.appendCalls == 0 {
+		return
+	}
+
+	// set the title and the add/remove legend
+	title := strings.ToUpper(d.title)
+	add := "+add"
+	remove := "-remove"
+	if d.useColor {
+		title = colorTitle.Sprint(title)
+		add = colorTotalAdd.Sprint(add)
+		remove = colorTotalRemove.Sprint(remove)
+	}
+	fmt.Fprintf(d.w, "%s    %s | %s | unchanged\n", title, add, remove)
+
+	d.setFooter()
+	d.writer.Render()
```

Comment:
There was a `fmt.Fprintln(d.w)` here in the `influxdb` code...was that not needed here?  ... or there?

Reply:
I'd originally ported it assuming that `Render()` wasn't writing a trailing newline, but found from testing that it wasn't needed. Maybe in the `apply` output it's nice to have blank lines between each output; in that case, I'll add the println's back there.